databroker/src/glob.rs

```diff
@@ -19,6 +20,73 @@ pub enum Error {
     RegexError,
 }
 
+#[derive(Debug)]
+pub enum MatchError {
+    MatchError,
+}
+
+#[derive(Debug)]
+pub struct Matcher {
+    glob_pattern: String,
+}
+
+impl Matcher {
+    pub fn new(glob_pattern: &str) -> Result<Self, MatchError> {
+        // Check if the pattern is valid
+        if is_valid_pattern(glob_pattern) {
+            Ok(Matcher {
+                glob_pattern: Matcher::to_glob_string(glob_pattern),
+            })
+        } else {
+            Err(MatchError::MatchError)
+        }
+    }
+
+    pub fn is_match(&self, path: &str) -> bool {
+        glob_match(&self.glob_pattern, path)
+    }
+
+    // Only support branches like Vehicle.Cabin.Sunroof but not like **.Sunroof
+    pub fn is_branch_matcher(&self) -> bool {
+        !self.glob_pattern.starts_with("**") && !self.glob_pattern.ends_with("/**")
+    }
+
+    pub fn transform_pattern_path_to_branch(&mut self) {
+        self.glob_pattern += "/**";
+    }
```

Comment:
Does `transform_pattern_path_to_branch` really belong in glob.rs. It's seems to be a very use case specific function that is better handled at the call site. If we want to match using a different glob, just create a new one:
```rust
glob::Matcher::new(old_pattern + "/**")
```